README.md
[![Apache 2](http://img.shields.io/badge/license-Apache%202-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)

# Plesk WordPress Edition

This repository contains all you need to build and customize your personal preconfigured WordPress Server solution with Plesk. The included whitepaper describes the solution in detail and highlights why you should offer it to your customers.

![](https://raw.githubusercontent.com/plesk/ext-welcome-guide/master/src/_meta/screenshots/2.png)

## Description

  Here at Plesk we strive to Simpify the Lives of Web Professionals. This includes our Partners. This cookbook will install and configure Plesk Core, Modules and Bundle Extensions. It will also enable and configure security features and prepare your VM to be used as a VM template. 

## Requirements

 * Contract with Plesk to be able to retrieve Plesk Licenses from the Key Administrator Server (KA)
 * Provide Plesk Key with Proper Plesk Exension keys associated in cookbook script

 or

  * A Plesk license with the [WordPress Pack](https://ext.plesk.com/packages/f8afea30-b1e9-45f5-ab71-437a0ce7c817-offer-wordpress-feature-pack)
 
## Deploying Plesk with WordPress Edition Cookbook

1. Install one of the supported Linux Operating Systems ( https://docs.plesk.com/release-notes/onyx/hardware-requirements ) following hardware specs ( https://docs.plesk.com/release-notes/onyx/hardware-requirements/ )

2. Shell to server as root

3. Download and Edit Variables noted in install_wordpress_edition.sh

4. Execute command 

   $ chmod +x install install_wordpress_edition.sh

5. Execute command 

   $ ./install_wordpress_edition.sh <YOUR_ACTIVATION_KEY>

### Using prebuild Plesk Images

tbd

## Support

Found a bug or experienced an issue? Please open an [issue on Github](https://github.com/plesk/wordpress-edition/issues)!


CHANGES.md
# 1.2.1
* [*] Fixed bug with plesk_ui
* [*] Changed default plesk_ui to Service Provider View

# 1.2
* [*] Replaced old Welcome Extension with the new Welcome Guide

# 1.1
* [*] Updated the cookbook for the latest version of the WordPress Edition
* [*] Replaced the Security Advisor with our new Advisor Extension
* [*] Replaced PHP5.6 with PHP7.2
* [*] Changed default behaviour of cloning to false
* [+] Added Extension installation for SEO Toolkit, Advisor, Uptime Robot and many more 
* [+] Added Solution Type for reporting purposes
* [+] Mentioned Feature Pack in README.md 
* [+] Added option to use the activation key as parameter
* [+] Added output of one-time login link if clone is set to off
* [-] Removed extensions no longer available in WordPress Edition like Opsani or Addendino
* [-] Removed Route 53 Extension
* [-] Removed no longer needed admin information

# 1.0.1

* [*] Updated link to the latest Welcome WordPress extension package (version 1.0.2)

# 1.0.0

* [+] First commit


Plesk-WordPress-Cookbook.sh
#!/bin/bash

#Plesk 
#Edit variables for Plesk Initialization

hostname="cp.domain.tst"
email="admin@test.tst"
passwd=CookBook123
aname=“admin”
company=Cookbook
phone="123-123-1234"
address=“123_street”
city=NY
state=NY
zip=12345
country=US
agreement=true
iptype=shared

#Plesk UI View - can be set to Service Provider View (spv) or Power User View (puv)
pleskui=puv

#Turn on Fail2Ban, yes or no, Keep in mind you need to provide temp license for initialization for this to work
fail2ban=yes

#Plesk Activation Code - provide proper license for initialization, it will be replaced after cloning
#leave as null if not providing key
activation_key=<put your activation code here>
#Turn on http2
http2=yes

######### Do not edit below this line ###################
#########################################################

#Download Plesk AutoInstaller

wget https://installer.plesk.com/plesk-installer



#Make Installed Executable

chmod +x ./plesk-installer




#Install Plesk with Required Components


./plesk-installer install plesk --preset Full --without resctrl sitebuilder drweb mailman java psa-vpn psa-fileserver watchdog magicspam health-monitor



#Initalize Plesk before Additional Configuration


echo "Starting initialization process of your Plesk server"
plesk bin init_conf --init -email $email -passwd $passwd -company $company -name $name -phone $phone -address $address -city $city -state $state -zip $zip -country $country -license_agreed $agreement -ip-type $iptype


#Install Plesk Activation Key if provided

if [ “$activation_key” != “null” ]; then

echo "Installing Plesk Activation Code"
plesk bin license --install $activation_key
echo "  "

fi



#Configure Service Provider View On
if [ "$pleskui" = "spv" ]; then
    echo "Setting to Service Provider View"
    /usr/sbin/plesk bin poweruser --off
    echo "  "
else
    echo "Setting to Power user View"
    /usr/sbin/plesk bin poweruser --on
    echo "  "
fi


#Make sure Plesk UI and Plesk Update ports are allowed

echo "Setting Firewall to allow Ports 8443 and 8447"
iptables -I INPUT -p tcp --dport 8443 -j ACCEPT
iptables -I INPUT -p tcp --dport 8447 -j ACCEPT
echo "  "

#Enable Modsecurity

echo "Turning on Modsecurity WAF Rules"
plesk sbin modsecurity_ctl --enable --enable-ruleset atomic
echo "  "


#Enable Fail2Ban and Jails


if [ “$fail2ban” = “yes” ]; then

echo "Configuring Fail2Ban and its Jails"
/usr/sbin/plesk bin ip_ban --enable
/usr/sbin/plesk bin ip_ban --enable-jails ssh
/usr/sbin/plesk bin ip_ban --enable-jails recidive
/usr/sbin/plesk bin ip_ban --enable-jails modsecurity
/usr/sbin/plesk bin ip_ban --enable-jails plesk-proftpd
/usr/sbin/plesk bin ip_ban --enable-jails plesk-postfix
/usr/sbin/plesk bin ip_ban --enable-jails plesk-dovecot
/usr/sbin/plesk bin ip_ban --enable-jails plesk-roundcube
/usr/sbin/plesk bin ip_ban --enable-jails plesk-roundcube
/usr/sbin/plesk bin ip_ban --enable-jails plesk-apache-badbot
/usr/sbin/plesk bin ip_ban --enable-jails plesk-panel
/usr/sbin/plesk bin ip_ban --enable-jails plesk-wordpress
echo "  "

fi


#Turn on http2

if [ “$http2” = “yes” ]; then
echo "Activating http2"
/usr/sbin/plesk bin http2_pref --enable
echo "  "
fi


#Install Bundle Extensions

echo "Installing Requested Plesk Extensions"
echo "Installing Route 53"
plesk bin extension --install-url https://ext.plesk.com/packages/ed1860ee-45c5-4e2b-b6b7-44e5da69dca5-route53/download?2.3-0
echo "  "
echo "Installing Security Advisor"
plesk bin extension --install-url https://ext.plesk.com/packages/6bcc01cf-d7bb-4e6a-9db8-dd1826dcad8f-security-advisor/download?1.4.0-4
echo "  "
echo "Installing Google Pagespeed Insights"
plesk bin extension --install-url https://ext.plesk.com/packages/3d2639e6-64a9-43fe-a990-c873b6b3ec66-pagespeed-insights/download?1.0.0-6
echo "  "
echo "Installing Addendio - WordPress Plugin and Themes"
plesk bin extension --install-url https://ext.plesk.com/packages/250589ff-8081-4c30-b9ca-66539e025c27-addendio-wordpress/download?1.2.0-73
echo "  "
echo "Installing Datagrid VCTR reliability and vulnerability scanner"
plesk bin extension --install-url https://ext.plesk.com/packages/e757450e-40a5-44e5-a35d-8c4c50671019-dgri/download?2.1-0
echo "  "
echo "Installing LetsEncrypt"
plesk bin extension --install-url https://ext.plesk.com/packages/f6847e61-33a7-4104-8dc9-d26a0183a8dd-letsencrypt/download?2.1.0-48
echo "  "
echo "Installing Plesk Migration Manager"
plesk bin extension --install-url https://ext.plesk.com/packages/bebc4866-d171-45fb-91a6-4b139b8c9a1b-panel-migrator/download?2.9.2-0
echo "  "


#Prepair do Clonging

echo "Setting Plesk Cloning feature."
plesk bin cloning --update -prepare-public-image true
echo "Plesk Initialization will be wiped on next boot. Ready for Cloning."


echo ""
echo "Your Plesk WordPress Bundle Image is complete."
echo "Thank you for using the Plesk Business Server Cookbook"
echo "  "
echo "  "


install_wordpress_edition.sh
#!/bin/bash

################################################################################
##          Installation script for Plesk WordPress Edition                   ##
################################################################################

# Edit variables for Plesk pre-configuration

email='admin@test.tst'
passwd='CookBook123'
name='admin'
agreement=true

# Plesk Activation Code - provide proper license for initialization, it will be replaced after cloning
# leave as null if not providing key
activation_key=$1

# Plesk UI View - can be set to Service Provider View (spv) or Power User View (puv)
plesk_ui=spv

# Turn on Fail2Ban, yes or no, Keep in mind you need to provide temp license for initialization for this to work
fail2ban=yes

# Turn on http2
http2=yes

# Turn on Cloning - Set to "on" if this it to make a Golden Image, set to "off" if for remote installation
clone=off

# Test to make sure all initialization values are set

if [[ -z $activation_key ]]; then
echo 'Please provide a proper Plesk Activation Code (Bundle License).'
  exit 1
fi

if [[ -z $email || -z $passwd || -z $name || -z $agreement ]]; then
  echo 'One or more variables are undefined. Please check your initialization values.'
  exit 1
fi

echo "Plesk initialization values are all assigned. We are good so far."
echo

######### Do not edit below this line ###################
#########################################################

# Download Plesk AutoInstaller

echo "Downloading Plesk Auto-Installer"
wget https://installer.plesk.com/plesk-installer
echo

# Make Installed Executable

echo "Making Plesk Auto-Installer Executable"
chmod +x ./plesk-installer
echo

# Install Plesk with Required Components

echo "Starting Plesk Installation"
./plesk-installer install plesk --preset Recommended --with fail2ban modsecurity spamassassin mailman psa-firewall pmm health-monitor

# If Ruby and NodeJS are needed then run install Plesk using the following command:
# ./plesk-installer install plesk --preset Recommended --with fail2ban modsecurity spamassassin mailman psa-firewall pmm health-monitor passenger ruby nodejs gems-preecho
echo
echo

# Initalize Plesk before Additional Configuration
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/init_conf-server-configuration.37843/

echo "Starting initialization process of your Plesk server"

plesk bin init_conf --init -email $email -passwd $passwd -name $name -license_agreed $agreement 

plesk bin settings --set solution_type="wordpress"
echo

# Install Plesk Activation Key if provided
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/license-license-keys.71029/

if [[ -n "$activation_key" ]]; then
  echo "Installing Plesk Activation Code"
  plesk bin license --install $activation_key
  echo
fi

# Configure Service Provider View On

if [ "$plesk_ui" = "spv" ]; then
    echo "Setting to Service Provider View"
    plesk bin poweruser --off
    echo
else
    echo "Setting to Power user View"
    plesk bin poweruser --on
    echo
fi

# Make sure Plesk UI and Plesk Update ports are allowed

echo "Setting Firewall to allow proper ports."
iptables -I INPUT -p tcp --dport 21 -j ACCEPT
iptables -I INPUT -p tcp --dport 22 -j ACCEPT
iptables -I INPUT -p tcp --dport 25 -j ACCEPT
iptables -I INPUT -p tcp --dport 80 -j ACCEPT
iptables -I INPUT -p tcp --dport 110 -j ACCEPT
iptables -I INPUT -p tcp --dport 143 -j ACCEPT
iptables -I INPUT -p tcp --dport 443 -j ACCEPT
iptables -I INPUT -p tcp --dport 465 -j ACCEPT
iptables -I INPUT -p tcp --dport 993 -j ACCEPT
iptables -I INPUT -p tcp --dport 995 -j ACCEPT
iptables -I INPUT -p tcp --dport 8443 -j ACCEPT
iptables -I INPUT -p tcp --dport 8447 -j ACCEPT
iptables -I INPUT -p tcp --dport 8880 -j ACCEPT

echo

# Enable Modsecurity
# https://docs.plesk.com/en-US/onyx/administrator-guide/server-administration/web-application-firewall-modsecurity.73383/

echo "Turning on Modsecurity WAF Rules"
plesk bin server_pref --update-web-app-firewall -waf-rule-engine on -waf-rule-set tortix -waf-rule-set-update-period daily -waf-config-preset tradeoff
echo

# Enable Fail2Ban and Jails
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/ip_ban-ip-address-banning-fail2ban.73594/

if [ "$fail2ban" = "yes" ]; then
  echo "Configuring Fail2Ban and its Jails"
  plesk bin ip_ban --enable
  plesk bin ip_ban --enable-jails ssh
  plesk bin ip_ban --enable-jails recidive
  plesk bin ip_ban --enable-jails modsecurity
  plesk bin ip_ban --enable-jails plesk-proftpd
  plesk bin ip_ban --enable-jails plesk-postfix
  plesk bin ip_ban --enable-jails plesk-dovecot
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-apache-badbot
  plesk bin ip_ban --enable-jails plesk-panel
  plesk bin ip_ban --enable-jails plesk-wordpress
  plesk bin ip_ban --enable-jails plesk-apache
  plesk bin ip_ban --enable-jails plesk-horde
  echo
fi

# Turn on http2
# https://docs.plesk.com/en-US/onyx/administrator-guide/web-servers/apache-and-nginx-web-servers-linux/http2-support-in-plesk.76461/

if [ "$http2" = "yes" ]; then
  echo "Activating http2"
  /usr/sbin/plesk bin http2_pref --enable
  echo
fi

# Install Bundle Extensions
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/extension-extensions.71031/

echo "Installing Requested Plesk Extensions"
echo
echo "Installing SEO Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/2ae9cd0b-bc5c-4464-a12d-bd882c651392-xovi/download
echo
echo "Installing BoldGrid"
plesk bin extension --install-url https://ext.plesk.com/packages/e4736f87-ba7e-4601-a403-7c82682ef07d-boldgrid/download
echo
echo "Installing Backup to Cloud extensions"
plesk bin extension --install-url https://ext.plesk.com/packages/9f3b75b3-d04d-44fe-a8fa-7e2b1635c2e1-dropbox-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/52fd6315-22a4-48b8-959d-b2f1fd737d11-google-drive-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/8762049b-870e-47cb-ba14-9f055b99b508-s3-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/a8e5ad9c-a254-4bcf-8ae4-5440f13a88ad-one-drive-backup/download
echo
echo "Installing Speed Kit"
plesk bin extension --install-url https://ext.plesk.com/packages/11e1bf5f-a0df-48c6-8761-e890ff4e906c-baqend/download
echo
echo "Installing Revisium Antivirus for Websites"
plesk bin extension --install-url https://ext.plesk.com/packages/b71916cf-614e-4b11-9644-a5fe82060aaf-revisium-antivirus/download
echo
echo "Installing Google Pagespeed Insights"
plesk bin extension --install-url https://ext.plesk.com/packages/3d2639e6-64a9-43fe-a990-c873b6b3ec66-pagespeed-insights/download
echo
echo "Installing Uptime Robot"
plesk bin extension --install-url https://ext.plesk.com/packages/7d37cfde-f133-4085-91ea-d5399862321b-uptime-robot/download
echo
echo "Installing Sucuri Site Scanner"
plesk bin extension --install-url https://ext.plesk.com/packages/2d5b423b-9104-40f2-9286-a75a6debd43f-sucuri-scanner/download
echo 
echo "Installing Domain Connect"
plesk bin extension --install-url https://ext.plesk.com/packages/3a36f828-e477-4600-be33-48c21e351c9a-domain-connect/download
echo
echo "Installing Welcome Guide"
plesk bin extension --install-url https://ext.plesk.com/packages/39eb8f3d-0d9a-4605-a42a-c37ca5809415-welcome/download
echo
echo "Enabling Welcome Guide for the Plesk WordPress Edition"
plesk ext welcome --select -preset wordpress
echo 


echo "Enabling Welcome Guide for the Plesk WordPress Edition"
plesk ext welcome --select -preset wordpress
echo

# Prepair for Cloning
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/cloning-server-cloning-settings.71035/

if [ "$clone" = "on" ]; then
	echo "Setting Plesk Cloning feature."
	plesk bin cloning --update -prepare-public-image true -reset-lincese true -skip-update true
	echo "Plesk initialization will be wiped on next boot. Ready for Cloning."
else
  echo "Here is your login"
  plesk login
fi

echo
echo "Your Plesk WordPress Edition is complete."
echo "Thank you for using the WordPress Edition Cookbook"
echo


install_wordpress_server.sh
#!/bin/bash

################################################################################
##          Installation script for Plesk WordPress Server                    ##
################################################################################

# Edit variables for Plesk pre-configuration

hostname='cp.domain.tst'
email='admin@test.tst'
passwd='CookBook123'
name='admin'
agreement=true
ip_type=shared

# Plesk Activation Code - provide proper license for initialization, it will be replaced after cloning
# leave as null if not providing key
activation_key=$1


# Plesk UI View - can be set to Service Provider View (spv) or Power User View (puv)
plesk_ui=puv

# Turn on Fail2Ban, yes or no, Keep in mind you need to provide temp license for initialization for this to work
fail2ban=yes

# Turn on http2
http2=yes

# Turn on Cloning - Set to "on" if this it to make a Golden Image, set to "off" if for remote installation
clone=off

# Test to make sure all initialization values are set

if [[ -z $activation_key ]]; then
echo 'Please provide a proper Plesk Activation Code (Bundle License).'
  exit 1
fi

if [[ -z $hostname || -z $email || -z $passwd || -z $name || -z $agreement || -z $ip_type ]]; then
  echo 'One or more variables are undefined. Please check your initialization values.'
  exit 1
fi

echo "Plesk initialization values are all assigned. We are good so far."
echo

######### Do not edit below this line ###################
#########################################################

# Download Plesk AutoInstaller

echo "Downloading Plesk Auto-Installer"
wget https://installer.plesk.com/plesk-installer
echo

# Make Installed Executable

echo "Making Plesk Auto-Installer Executable"
chmod +x ./plesk-installer
echo

# Install Plesk with Required Components

echo "Starting Plesk Installation"
./plesk-installer install plesk --preset Recommended --with panel bind fail2ban l10n pmm mysqlgroup roundcube kav spamassassin postfix dovecot proftpd awstats modsecurity mod_fcgid webservers php7.2 php7.1 config-troubleshooter psa-firewall heavy-metal-skin letsencrypt 
echo
echo

# Initalize Plesk before Additional Configuration
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/init_conf-server-configuration.37843/

echo "Starting initialization process of your Plesk server"
plesk bin init_conf --init -email $email -passwd $passwd -name $name -hostname $hostname -license_agreed $agreement -ip-type $ip_type 
plesk bin settings --set solution_type="wordpress"
echo


# Install Plesk Activation Key if provided
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/license-license-keys.71029/

if [[ -n "$activation_key" ]]; then
  echo "Installing Plesk Activation Code"
  plesk bin license --install $activation_key
  echo
fi

# Configure Service Provider View On

if [ "$pleskui" = "spv" ]; then
    echo "Setting to Service Provider View"
    plesk bin poweruser --off
    echo
else
    echo "Setting to Power user View"
    plesk bin poweruser --on
    echo
fi

# Make sure Plesk UI and Plesk Update ports are allowed

echo "Setting Firewall to allow proper ports."
iptables -I INPUT -p tcp --dport 21 -j ACCEPT
iptables -I INPUT -p tcp --dport 22 -j ACCEPT
iptables -I INPUT -p tcp --dport 25 -j ACCEPT
iptables -I INPUT -p tcp --dport 80 -j ACCEPT
iptables -I INPUT -p tcp --dport 110 -j ACCEPT
iptables -I INPUT -p tcp --dport 143 -j ACCEPT
iptables -I INPUT -p tcp --dport 443 -j ACCEPT
iptables -I INPUT -p tcp --dport 465 -j ACCEPT
iptables -I INPUT -p tcp --dport 993 -j ACCEPT
iptables -I INPUT -p tcp --dport 995 -j ACCEPT
iptables -I INPUT -p tcp --dport 8443 -j ACCEPT
iptables -I INPUT -p tcp --dport 8447 -j ACCEPT
iptables -I INPUT -p tcp --dport 8880 -j ACCEPT

echo

# Enable Modsecurity
# https://docs.plesk.com/en-US/onyx/administrator-guide/server-administration/web-application-firewall-modsecurity.73383/

echo "Turning on Modsecurity WAF Rules"
plesk bin server_pref --update-web-app-firewall -waf-rule-engine on -waf-rule-set tortix -waf-rule-set-update-period daily -waf-config-preset tradeoff
echo

# Enable Fail2Ban and Jails
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/ip_ban-ip-address-banning-fail2ban.73594/

if [ "$fail2ban" = "yes" ]; then
  echo "Configuring Fail2Ban and its Jails"
  plesk bin ip_ban --enable
  plesk bin ip_ban --enable-jails ssh
  plesk bin ip_ban --enable-jails recidive
  plesk bin ip_ban --enable-jails modsecurity
  plesk bin ip_ban --enable-jails plesk-proftpd
  plesk bin ip_ban --enable-jails plesk-postfix
  plesk bin ip_ban --enable-jails plesk-dovecot
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-roundcube
  plesk bin ip_ban --enable-jails plesk-apache-badbot
  plesk bin ip_ban --enable-jails plesk-panel
  plesk bin ip_ban --enable-jails plesk-wordpress
  echo
fi

# Turn on http2
# https://docs.plesk.com/en-US/onyx/administrator-guide/web-servers/apache-and-nginx-web-servers-linux/http2-support-in-plesk.76461/

if [ "$http2" = "yes" ]; then
  echo "Activating http2"
  /usr/sbin/plesk bin http2_pref --enable
  echo
fi

# Install Bundle Extensions
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/extension-extensions.71031/

echo "Installing Requested Plesk Extensions"
echo
echo "Installing WordPress Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/00d002a7-3252-4996-8a08-aa1c89cf29f7-wp-toolkit/download
echo "Installing SEO Toolkit"
plesk bin extension --install-url https://ext.plesk.com/packages/2ae9cd0b-bc5c-4464-a12d-bd882c651392-xovi/download
echo
echo "Installing Advisor"
plesk bin extension --install-url https://ext.plesk.com/packages/bbf16bc7-094e-4cb3-8b9c-32066fc66561-advisor/download
echo
echo "Installing BoldGrid"
plesk bin extension --install-url https://ext.plesk.com/packages/e4736f87-ba7e-4601-a403-7c82682ef07d-boldgrid/download
echo
echo "Installing Backup to Cloud Pro"
plesk bin extension --install-url https://ext.plesk.com/packages/9f3b75b3-d04d-44fe-a8fa-7e2b1635c2e1-dropbox-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/52fd6315-22a4-48b8-959d-b2f1fd737d11-google-drive-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/8762049b-870e-47cb-ba14-9f055b99b508-s3-backup/download
plesk bin extension --install-url https://ext.plesk.com/packages/a8e5ad9c-a254-4bcf-8ae4-5440f13a88ad-one-drive-backup/download
echo
echo "Installing Speed Kit"
plesk bin extension --install-url https://ext.plesk.com/packages/11e1bf5f-a0df-48c6-8761-e890ff4e906c-baqend/download
echo "Installing Revisium Antivirus for Websites"
plesk bin extension --install-url https://ext.plesk.com/packages/b71916cf-614e-4b11-9644-a5fe82060aaf-revisium-antivirus/download
echo
echo "Installing Google Pagespeed Insights"
plesk bin extension --install-url https://ext.plesk.com/packages/3d2639e6-64a9-43fe-a990-c873b6b3ec66-pagespeed-insights/download
echo
echo "Installing Uptime Robot"
plesk bin extension --install-url https://ext.plesk.com/packages/7d37cfde-f133-4085-91ea-d5399862321b-uptime-robot/download
echo
echo "Installing Sucuri Site Scanner"
plesk bin extension --install-url https://ext.plesk.com/packages/2d5b423b-9104-40f2-9286-a75a6debd43f-sucuri-scanner/download
echo 
echo "Installing LetsEncrypt"
plesk bin extension --install-url https://ext.plesk.com/packages/f6847e61-33a7-4104-8dc9-d26a0183a8dd-letsencrypt/download
echo
echo "Installing Plesk Migration Manager"
plesk bin extension --install-url https://ext.plesk.com/packages/bebc4866-d171-45fb-91a6-4b139b8c9a1b-panel-migrator/download
echo
echo "Installing Welcome Extension"
plesk bin extension --install-url https://github.com/plesk/ext-welcome-wordpress/releases/download/v1.0.2/ext-welcome-wp_v1.0.2-11.zip
echo


# Prepair for Cloning
# https://docs.plesk.com/en-US/onyx/cli-linux/using-command-line-utilities/cloning-server-cloning-settings.71035/


if [ "$clone" = "on" ]; then
	echo "Setting Plesk Cloning feature."
	plesk bin cloning --update -prepare-public-image true
	echo "Plesk initialization will be wiped on next boot. Ready for Cloning."
else
  echo "Here is your login"
  plesk login
fi

echo
echo "Your Plesk WordPress Server image is complete."
echo "Thank you for using the WordPress Server Cookbook"
echo
echo
